pkg/cache/unavailableofferings.go

```diff
@@ -48,10 +49,18 @@ func NewUnavailableOfferings() *UnavailableOfferings {
 
 // IsUnavailable returns true if the offering appears in the cache
 func (u *UnavailableOfferings) IsUnavailable(instanceType, zone, capacityType string) bool {
+	if _, ok := u.cache.Get(u.key("", "", v1beta1.CapacityTypeSpot)); ok && capacityType == v1beta1.CapacityTypeSpot {
```

Comment:
Checking capacityType first would be faster. Can also pre-compute the key, once. (To minimize performance impact we introduce here. Yes, slight, but called frequently ..) Otherwise 👍

Reply:
Is there a way to write 

``` 
if capacityType == v1beta1.CapacityTypeSpot {
		if _, ok := u.cache.Get(SpotUnavailableKey); ok {
			return true
		}
	}
``` 
As 
```
if capacityType == v1beta1.CapacityTypeSpot && _, ok := u.cache.Get(SpotUnavailableKey); ok {
```


```
The Go programming language does not allow declarations
 after &&. This is because && is a logical operator
 that returns true if both of its operands are true, and false otherwise. If a declaration were allowed after &&, it would be possible for the declaration to be executed before the && operator is evaluated, which could lead to sadness
For example, consider the following code
if true && var := 1; var > 0 {
  // This code will be executed, even if var is not initialized.
}

In this example, the variable var is declared after the && operator. This means that the declaration will be executed before the && operator is evaluated. If var is not initialized before the && operator is evaluated, it will be equal to the zero value of its type, which is 0. This means that the && operator will evaluate to false, and the code inside the if statement will not be executed.
To avoid this problem, declarations must be made before the && operator. This ensures that the declaration is executed before the && operator is evaluated, and that the variable is initialized before it is used.
```


I wanted to write the cpaaciyy type check before, but encountered this problem. I can use the first syntax to accomplish this just wondering if there is a better way i am missing.

Reply:
Not that I know of; I would not worry about it

Reply:
sg